### PR TITLE
Ghc 9.2 compat

### LINKED ISSUE #363: Remove Option
It is deprecated in `4.16`, we should remove it after moving to another version in relude.

### PR DESCRIPTION
**Pull request by TeofilC**
Resolves #363 and #388 
Depends on #400 .

This is #387 resurrected.
I think I've managed to fix the doctest failures, and added a 9.2.2 job to CI.
The version of cabal-install had to be bumped on CI otherwise it imposes unsatisfiable constraints from the Cabal library.

### HLint

- [X] I've changed the exposed interface (add new reexports, remove reexports, rename reexported things, etc.).
  - [X] I've updated [`hlint.dhall`](https://github.com/kowainik/relude/blob/master/hlint/hlint.dhall) accordingly to my changes (add new rules for the new imports, remove old ones, when they are outdated, etc.).
  - [X] I've generated the new `.hlint.yaml` file (see [this instructions](https://github.com/kowainik/relude#generating-hlintyaml)).

### General

- [X] I've updated the [CHANGELOG](https://github.com/kowainik/relude/blob/master/CHANGELOG.md) with the short description of my latest changes.
- [X] All new and existing tests pass.
- [ ] I keep the code style used in the files I've changed (see [style-guide](https://github.com/kowainik/org/blob/master/style-guide.md#haskell-style-guide) for more details).
- [ ] I've used the [`stylish-haskell` file](https://github.com/kowainik/relude/blob/master/.stylish-haskell.yaml).
- [ ] My change requires the documentation updates.
  - [ ] I've updated the documentation accordingly.
- [ ] I've added the `[ci skip]` text to the docs-only related commit's name.

### CODE REVIEW

File: relude.cabal [hunk 1]
```diff
@@ -104,7 +104,7 @@ source-repository head
   location: git@github.com:kowainik/relude.git
```
Comment (vrom911) [inline relude.cabal:104]: Can you also please add GHC-9.2 to the `tested-with` above? 🙂

**Review by vrom911 — APPROVED** [review]
Thanks a lot, @TeofilC !
This looks great 👍🏼

File: relude.cabal (hunk 1 above)
Comment (TeofilC) [reply relude.cabal:105]: Good catch, should be fixed by 23feee16dae03e0796ac3cb2e318c943d906ca22

**Review by chshersh — APPROVED** [review]
This looks great! Awesome work @TeofilC 👏🏻 🔝

**Pull request merged**